src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -388,6 +392,13 @@ StartRemoteTransactionPrepare(struct MultiConnection *connection)
 
 	Assign2PCIdentifier(connection);
 
+	/* log transactions to workers in pg_dist_transaction */
```

Comment:
Misunderstood the comment. I looked into moving FindWorkerNode to StartConnectionEstablishment where the MultiConnection is initialised, but that code path is called from a few different contexts such as the task tracker background worker which made that a bit impractical. For now I'll stick to the current approach, which usually only involves a hash look-up.